File: testdata/hs/sample.hs
greet :: String -> String
greet name =
  "Hello, " ++ name ++ "!"

add :: Int -> Int -> Int
add x y = x + y

data Color
  = Red
  | Green
  | Blue

data Point = Point
  { px :: Double
  , py :: Double
  }

class Shape a where
  area :: a -> Double
  perimeter :: a -> Double

File: testdata/php/sample.php
<?php

function greet(string $name): string {
  return "Hello, " . $name . "!";
}

function add(int $x, int $y): int {
  return $x + $y;
}

class Point {
  public float $x;
  public float $y;

  public function __construct(float $x, float $y) {
    $this->x = $x;
    $this->y = $y;
  }

  public function distance(): float {
    return sqrt($this->x ** 2 + $this->y ** 2);
  }
}

interface Shape {
  public function area(): float;
}

trait Greetable {
  public function hello(): string {
    return "Hello from " . get_class($this);
  }
}
